### ISSUE TITLE
Throwing an exception in a mocked method does not count that method as invoked

### ISSUE THREAD

**Issue by rikh42** [Codeception/AspectMock#69]
``` php
// Mock a method in a class. Use a closure to throw an exception
$apiMock = test::double(MyExample::class, [
    'someMethod' => function () {
        // Fake up the function failing by throwing one of the exceptions it normally uses for an error
        throw new BrokenStuffException('It broke');
    }
]);

try {
    $m = new MyExample;
    $m->someMethod();
} catch (BrokenStuffException $e) {
//...
}

// This fails, saying someMethod was not invoked
$apiMock->verifyInvoked('someMethod');
```

In the example above, everything works - the closure is called, the exception thrown, the code being tested handles the exception like normal. 

However, Aspect Mock does not appear to have counted the call to 'someMethod' anywhere and thinks it was never invoked.

**Comment by eggertj12** [comment]
I'm getting started with this excellent tool and hitting the same issue. Lack the skills to go through the source and find anything useful and my googling has returned nothing but this issue.

Any news on this?

**Comment by GuillaumeDievart** [comment]
up ! Same issue here

**Comment by Naktibalda** [comment]
I merged pull request.

**Comment by Naktibalda** [comment]
Released as 3.1.0
